### PR TITLE
Automatic correction support

### PR DESCRIPTION
**Pull request by Feuermagier**
### Checklist
- [x] I tested *all* changes and their related features locally or with a test instance of Artemis.
- [x] I documented the Java code using JavaDoc style.
- [ ] I documented the changes in the Documentation (/docs).

### Motivation and Context
This PR exists mainly to trigger a build of the plugin with Autograder integration.

Automatic correction may contribute to making the assessment process more fair and reduce the workload of tutors.

### Description
Include the Autograder and execute it automatically on each submission. The implementation is currently extremely dirty and more or less hardcoded for the current plugin configuration. Because of problems with Maven dependencies and Eclipse the Autograder is currently included as a Jar file which has to be manually updated whenever the Autograder itself is updated.

### Steps for Testing
Start a new assessment, wait a few seconds and observe the magic of automatic correction!

### CODE REVIEW

File: bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/AutograderUtil.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+							String.format("Autograder found %d issues. Please check that there are no false-positives.", annotations.size())));
+				}
+			} catch (Exception ex) {
+				LOG.warn(ex.getMessage());
+			}
+		}).schedule();
+	}
+
+	private static Optional<IMistakeType> mapAnnotation(IAssessmentController assessmentController, AutograderAnnotation annotation) {
+		String id = switch (annotation.type()) {
```
[the remark this answers is not in the document]
Comment (Feuermagier) [reply bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/AutograderUtil.java:110]: Yeah, of course, that was only a temporary solution to get the integration working for the current tasks :P. I think we could make this part of the exercise config, maybe with an additional array per penalty type that contains all mistake types that should be mapped to this penalty.

**Comment by Luro02** [comment]
It would be very useful if the error message contains a suggestion on where one can find the logs. For example: `"Autograder failed. Please assess the submission normally. Additional information can be found in the Eclipse log. The log can be accessed through Window > Show View > Other > General > Error Log."`

**Comment by dfuchss** [comment]
@Feuermagier I've pushed a concept to call the autograder directly. Unfortunately, eclipse crashes directly after it's finished (at least I think so). Therefore, I reverted the changes afterwards. Maybe it's a starting point to integrate the autograder directly.

**Comment by Feuermagier** [comment]
@dfuchss Can you reopen this PR? There are several changes on my fork, including the use of a config file that specifies the mapping between autograder problems and grading tool mistakes.

I've also included an option to automatically download the autograder JAR from Github (always using the latest release for every new submission, and using a cached JAR if possible). While this is not exactly the solution that we have talked about, it was the simplest to implement for me. However, a local JDK installation and java in the PATH is still required.

**Comment by dfuchss** [comment]
Thanks a lot for your work. I'll have a look at it asap :)

**Review by dfuchss — APPROVED** [review]
Nice work. For me that works well :)

Could you slightly adopt the Documentation ? E.g., add some details about the configuration of the autograder? Otherwise could you create an issue for that. I think the documentation of the feature might be very useful :)

**Pull request merged**